### DISCUSSION TITLE
Birdseye with substreams

### DISCUSSION THREAD

**Discussion by mcarson75** [blakeblackshear/frigate#12559] [General]
Is there an easy way to use substreams for birdseye view?  I have main streams and substreams defined in go2rtc, and camera is using the main stream for record and the substream for detect.  Birdseye is using all the main streams, and it doesn't quite seem smooth - only updates every few seconds and I get a spinning circle on refresh.

Is there an easy way to configure birdseye to use the substream instead so it is not having to transcode the full res video down?  I think I could define additional cameras that only use the substream and do it that way, but I was hoping to avoid that if possible.  Is there something like a `role: birdseye` for the specific stream?

**Comment by NickM-27** [comment]
> Birdseye is using all the main streams, and it doesn't quite seem smooth - only updates every few seconds and I get a spinning circle on refresh.

that is not correct, birdseye uses whatever stream is used for detect. This is because birdseye can only work with decoded streams, so it must work with detect which is decoded for motion / object detection.

**Reply by mcarson75** [reply]
Hmm, this doesn't seem to be the case in my experience.  I've got one Frigate instance just for doing recording and have 
```
    detect:
      enabled: False
```
on every camera.  But I still have a birdseye view.  So if birdseye can only work with decoded streams and it only gets decoded when detect is used, how is birdseye working?

**Reply by hawkeye217** [reply]
Frigate always decodes streams. Enabling or disabling object detection doesn't change that behavior.

**Reply by mcarson75** [reply]
So if detect is disabled, is there another way to force Birdseye to use the substream?

**Reply by NickM-27** [reply]
you can still assign the detect role to the sub stream and that is what will be used for birdseye

**Reply by mcarson75** [reply]
Hmm, I don't think I will do that on this instance since I do not want to do detection on this box.  I have another Frigate instance doing detection and analysis where I am using the substream for detection (since that box has a Coral), but it also seems to be using the main stream for birdseye based on the appearance of the timestamp (the substream on my camera uses a different font).  I am doing some investigation to try to find out for sure.

**Reply by NickM-27** [reply]
It is recommended to do it either way, otherwise you are decoding the recording stream which is just more CPU usage for no purpose. Using the detect role does not mean object detection is enabled

**Reply by mcarson75** [reply]
Ah okay, so I can add the detect role to the substream but leave detect disabled?  I will give that a try.

**Reply by NickM-27** [reply]
yes

**Comment by hawkeye217** [comment]
Also good to note that birdseye runs at whatever fps your `detect` role runs at, capped at 10fps.
